### PR TITLE
Implement 404 and Error pages

### LINKED ISSUE #246: Implement 404 and Error Page
**Description**
Create user-friendly 404 (Not Found) and Error pages for the website to handle requests for non-existent pages or resources.

**Acceptance Criteria**
- Design visually appealing 404 and Error pages that matches the website's overall style
- Include a clear "Page Not Found" message
- Add a navigation menu or link to return to the homepage
- Ensure the page is responsive and displays correctly on desktop and mobile devices
- Test the 404 and Error pages across major browsers (Chrome, Firefox, Safari)

### PR DESCRIPTION
**Pull request by chrisr0892**
##### Implemented a 404 page 
- Added button that routes back to home page
- Ensured the 404 page displays correctly on different screen sizes and across major browsers
- Styled the 404 page according to the design guidelines (styles file created under client/styles)
- removed separate Not-Found-Page folder to consolidate error files under error-pages folder.

##### Implemented Error page
- Styled the ErrorPage page according to the design guidelines (styles file created under client/styles)
- Added optional children props to MainLayout and UnauthorizedLayout so that ErrorPage will render with consistent styling.
- added ternary to ErrorElement in index.tsx

Fixes #246 

## Type of change

- [x] ✨  New feature (non-breaking change which adds functionality)

## How Can this be tested? Testing Plan to review this PR

- [x] Where in the app does this change take place?
- client/features/error-pages/NotFoundPage.tsx
- [x] What are the steps to test this change?
- run `npm run dev`, then try navigating to a route that does not exist.
- [x] What is the expected output?
- 404 page should render, clicking on button should reroute to homepage

- [x] Where in the app does this change take place?
- client/features/error-pages/ErrorPage.tsx
- [x] What are the steps to test this change?
- run `npm run dev`, then add 'throw new Error()' to a component before the return statement
- [x] What is the expected output?
- Error page should render, clicking on button should reroute to homepage

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] My changes generate no new warnings.
- [x] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [x] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [x] I have ensured that my pull request title is descriptive.

### CODE REVIEW

File: client/features/not-found-page/NotFoundPage.tsx
Comment (kinskiwu) [inline client/features/not-found-page/NotFoundPage.tsx:1]: Thanks for the good work! THe RouteLInk works very smoothly. 

Sorry that I totally forgot to link the design mockup in the ticket. Please find the 404 error mockup under 'Lo-fi Mockups - MVP' in this link https://www.figma.com/design/9QKsvDQ9jivgA9XfS0oqPd/Design-System---Linkta?node-id=691-3754&t=LIJGGXunTgU93zzD-0 and update the page.

File: client/styles/layout/NotFoundPage.module.css
Comment (kinskiwu) [inline client/styles/layout/NotFoundPage.module.css:1]: The styling looks pretty good! Would be nicer if the content is centered in the viewpoint. You may also want to lower the opacity of the text and button the same way as in the mockup.

**Review by kinskiwu — COMMENTED** [review]
Great work on your first ticket! Just some small changes related to design and styling. Also you may want to rebase your branch and resolve any merge conflicts (let us know if you need any help with that because it could be a bit tedious).

File: client/styles/layout/NotFoundPage.module.css
Comment (kinskiwu) [inline client/styles/layout/NotFoundPage.module.css:1]: I think this file should be outside of `/layout`

**Review by kinskiwu — APPROVED** [review]
Nice work! Pls remove the unused png inside /public then this pr is good to merge.

**Review by relandboyle — CHANGES_REQUESTED** [review]
@chrisr0892 This looks really good. Apologies in advance for all the things I'm going to ask you to change...

In the diffs for this PR it looks like you created both the ErrorPage component and stylesheet and the NotFoundPage component and stylesheet. Is that accurate? Or is that a side effect of some Git action that didn't go according to plan? As Kinski commented, please reach out if you need help merging or rebasing to get your branch prepared for review.

In the NotFoundPage.tsx component, I think you could achieve the same styling without the additional 'container' div. Also, in a Material UI app we should always use Material UI elements when possible. That means Box rather than Div, and Typography rather than h1, h2, and paragraph. I think you'll quickly grasp the approach if you review the MUI docs page for [Typography](https://mui.com/material-ui/react-typography/). 

In the NotFoundPage.module.css file, at the top I see a commented-out rule that looks like it was meant to be a temporary / experimental rule. I'm also noticing that you have some rules relating to headings and paragraphs which are not being applied. Those can be removed, and then when you implement the Typography elements you can use css modules if you need to override default styles.

**Review by relandboyle — APPROVED** [review]

**Pull request merged**